test/metabase/server/routes/index_test.clj

```diff
@@ -45,5 +45,32 @@
             "translations" {"" {"Metabase" {"msgid" "Metabase", "msgstr" ["Metabase"]}}}}
            (some->
             (binding [i18n/*user-locale* "en"]
-              (#'index/load-localization))
+              (#'index/load-localization nil))
             json/parse-string)))))
+
+(deftest override-localization-test
+  (testing "a valid override is honored no matter what the user locale is"
+    (is (= {"charset"      "utf-8"
+            "headers"      {"mime-version"              "1.0"
+                            "content-type"              "text/plain; charset=UTF-8"
+                            "content-transfer-encoding" "8bit"
+                            "x-generator"               "POEditor.com"
+                            "project-id-version"        "Metabase"
+                            "language"                  "es"
+                            "plural-forms"              "nplurals=2; plural=(n != 1);"}
+            "translations" {"" {"Your database has been added!" {"msgstr" ["¡Tu base de datos ha sido añadida!"]}}}}
+           (some->
+            (binding [i18n/*user-locale* "xx"]
+              (#'index/load-localization "es"))
+            json/parse-string
+            (update "translations" select-keys [""])
+            (update-in ["translations" ""] select-keys ["Your database has been added!"])))))
+
+  (testing "an invalid override causes a fallback to English"
+    (is (= {"headers"      {"language" "yy", "plural-forms" "nplurals=2; plural=(n != 1);"}
```

Comment:
The header has "language" "yy"... if we're actually defaulting to English, does it make sense to send "yy"? 

I think this was talked about before... perhaps it's to let the frontend know what language code failed? Not a blocker, I don't think.

Reply:
This [conversation](https://metaboat.slack.com/archives/CKZEMT1MJ/p1651173809142869) is exactly the thing why I asked specifically you.
Note that my new tests are in line with the existing ones. I think the logic is, that we are not "correcting" the frontend, we just give it the best translations we can.

Reply:
Yea, that sounds right. Sorry, it was a long thread and that detail got fuzzy in my mind. I'm happy with it :)